### ISSUE TITLE
Docker is not installed properly for Ubuntu 20.04 Focul AMI

### ISSUE THREAD

**Issue by prashantkalkar** [kubernetes/kops#9574]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

Kops version: v1.17.0

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

Kubernetes version: 1.14.10

**3. What cloud provider are you using?**

Cloud Provider: AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

Was just setting up one of the instance group. I use the CICD pipeline with the following commands:

kops toolbox template --template cluster-template.yaml --values values.yaml > cluster.yaml
kops replace --force -f cluster.yaml
kops update cluster
kops update cluster --yes
kops rolling-update cluster
kops rolling-update cluster --yes --master-interval=20s --node-interval=20s --post-drain-delay=10s --validation-timeout=30m

**5. What happened after the commands executed?**

One of the instance group AMI was set to : 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20200701

The IG instances never joined the cluster. The kops-configuration.service failed with error: 
W0714 21:55:44.857332    2027 docker.go:1385] Did not find docker package for focal amd64 18.06.3
Output: Failed to restart docker.service: Unit docker.socket not found.

**6. What did you expect to happen?**

The instances should get configured correctly and should join the cluster. 

It seems the main issue is missing the Docker package for 18.06.3. Let me know if more information is required.

**Comment by hakman** [comment]
If the main issue is the missing Docker package for `18.06.3`, I recommend using a newer version of Docker.
I think the latest 1.17 release is bundled with `19.03.11`.

**Comment by prashantkalkar** [comment]
I can change the docker version but I think it has to be done at the cluster level. I am trying out the ubuntu AMI just for a single instance group. Is it possible to change the docker version only for an instance group?

Currently, my cluster configuration does not specify any docker configuration. Anyway, the process should not fail with the default setup.

**Comment by hakman** [comment]
> I can change the docker version but I think it has to be done at the cluster level. I am trying out the ubuntu AMI just for a single instance group. Is it possible to change the docker version only for an instance group?

Docker version is a cluster level option and will impact all IGs.

> I think the process should not fail with the default setup. Currently, my cluster configuration does not specify any docker configuration.

Unfortunately, there are no `18.06.3` packages for Ubuntu 20.04:
https://download.docker.com/linux/ubuntu/dists/focal/pool/stable/amd64/

**Comment by prashantkalkar** [comment]
Sharing here as what I tried: 

I updated the cluster spec with docker version `19.03.11`. But the replace command failed with as unsupported version:

```
$ kops replace --force -f cluster.yaml
error: error replacing cluster: spec.docker.version: Unsupported value: "19.03.11": supported values: "17.03.2", "17.09.0", "18.03.1", "18.06.1", "18.06.2", "18.06.3", "18.09.3", "18.09.9", "19.03.4", "19.03.8"
```
So I changed the version to `19.03.8` and added an additional test instance group (IG) with Ubuntu AMI (I did not complete the rolling update for other IGs to avoid breaking anything).
 
It worked for the Ubuntu 20.04 Focal AMI (`099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20200701`)

Then to test the docker version change I added one more temp IG with default AMI which I am using for other IGs. 
(`kope.io/k8s-1.14-debian-stretch-amd64-hvm-ebs-2020-01-17`).

That IG nodes also worked without any issues. So it seems the docker version 19.03.8 is supported by other Ubuntu Image as well as default AMI for K8s version 1.14.10. I will now roll the change to all other IGs in the cluster. 

I think the AMI documentation should be updated with the supported docker versions for Ubuntu 20.04.

**Comment by hakman** [comment]
Please use Kops 1.17.1.

**Comment by prashantkalkar** [comment]
Ok. I tested with Kops `1.17.1` and docker version `19.03.11` is supported. I can confirm that the setup worked for me Ubuntu as well as default Kops AMI.

I think the only thing we should do is update the documentation to mention the minimum docker version required for Ubuntu AMI use.

**Comment by hakman** [comment]
> I think the only thing we should do is update the documentation to mention the minimum docker version required for Ubuntu AMI use.

Thanks for confirming @prashantkalkar .
Keep in mind that Docker 18.06 was released in 2018 and Ubuntu 20.04 was released in 2020. One should expect to be some limitations and do some research into compatibility.
That being said, feel free to add a docs PR about the minimum required Docker version for Ubuntu 20.04.

/close
